main.tf

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2020 Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+// readme.md created with https://terraform-docs.io/: terraform-docs markdown --sort=false ./ > ./readme.md
+
+// --- provider settings --- //
+terraform {
+  required_providers {
+    oci = {
+      source = "hashicorp/oci"
+    }
+  }
+}
+// --- provider settings  --- //
+
+// --- tenancy configuration --- //
+provider "oci" {
+  alias  = "service"
+  region = var.region
+}
+variable "tenancy_ocid" { }
+
+locals {
+  topologies = flatten(compact([var.host == true ? "host" : "", var.nodes == true ? "nodes" : "", var.container == true ? "container" : ""]))
+  domains    = jsondecode(file("${path.module}/default/resident/domains.json"))
+  segments   = jsondecode(file("${path.module}/default/network/segments.json"))
+}
+
+module "configuration" {
+  source         = "./default/"
+  providers = {oci = oci.service}
+  input = {
+    tenancy      = var.tenancy_ocid
+    class        = var.class
+    owner        = var.owner
+    organization = var.organization
+    solution     = var.solution
+    repository   = var.repository
+    stage        = var.stage
+    region       = var.region
+    osn          = var.osn
+  }
+  resolve = {
+    topologies = local.topologies
+    domains    = local.domains
+    segments   = local.segments
+  }
+}
+// --- tenancy configuration  --- //
+
+// --- operation controls --- //
+provider "oci" {
+  alias  = "home"
+  region = module.configuration.tenancy.region.key
+}
+module "resident" {
+  source = "github.com/ocilabs/resident"
```

Comment:
This should be updated to use the correct URL?

Reply:
Done